SU2_CFD/include/fluid/CFluidFlamelet.hpp

```diff
@@ -0,0 +1,146 @@
+ */
+
+#pragma once
+
+#include "../../Common/include/containers/CLookUpTable.hpp"
+#include "CFluidModel.hpp"
+
+class CFluidFlamelet final : public CFluidModel {
+ protected:
+  int rank;
+
+  unsigned short n_scalars;
+  unsigned short n_lookups;
+  unsigned short n_table_sources;
+  unsigned short n_user_scalars; /*!< \brief number of passive reactant species. */
+  unsigned short n_control_vars; /*!< \brief number of controlling variables. */
+
+  vector<string> table_scalar_names; /*!< \brief vector to store names of scalar variables.   */
+  vector<string> table_source_names; /*!< \brief vector to store names of scalar source variables.   */
+  vector<string> table_lookup_names; /*!< \brief vector to store names of look up variables.   */
+
+  vector<su2double> table_sources;
+
+  su2double mass_diffusivity; /*!< \brief local mass diffusivity of the mixture */
+  su2double molar_weight;     /*!< \brief local molar weight of the mixture */
+
+  vector<su2double> source_scalar;
+  vector<su2double> lookup_scalar;
+
+  CLookUpTable* look_up_table;
+
+  vector<string> varnames_TD;     /*!< \brief Lookup names for thermodynamic state variables. */
+  vector<su2double*> val_vars_TD; /*!< \brief References to thermodynamic state variables. */
+
+  vector<string> varnames_Sources;     /*!< \brief Lookup names for scalar source terms. */
+  vector<su2double*> val_vars_Sources; /*!< \brief References to scalar sources. */
+
+  vector<string> varnames_LookUp;     /*!< \brief Lookup names for passive lookup variables. */
+  vector<su2double*> val_vars_LookUp; /*!< \brief References to lookup variables. */
+
+ public:
+  CFluidFlamelet(CConfig* config, su2double value_pressure_operating);
+
+  ~CFluidFlamelet();
+
+  /*!
+   * \brief Set the thermodynamioc state
+   * \param[in] val_temperature - temperature
+   * \param[in] val_scalars - pointer to species mass fractions
+   */
+  void SetTDState_T(su2double val_temperature, const su2double* val_scalars = nullptr) override;
+
+  /*!
+   * \brief Set the reaction source terms for the transported species equations
+   * \param[in] val_scalars - pointer to species mass fractions
+   * \param[out] exit_code = error code
+   */
+  unsigned long SetScalarSources(const su2double* val_scalars);
+
+  /*!
+   * \brief Retrieve and set the lookup values for the species
+   * \param[in] val_scalars - pointer to species mass fractions
+   */
+  unsigned long SetScalarLookups(const su2double* val_scalars);
+
+  /*!
+   * \brief Get the total enthalpy from the tabulated temperature and species (inverse lookup)
+   * \param[in/out] enthalpy - total enthalpy
+   * \param[in] val_prog - progress variable
+   * \param[in] val_temp - temperature
+   * \param[out] exit_code = error code
+   */
+  unsigned long GetEnthFromTemp(su2double* enthalpy, const su2double val_prog, const su2double val_temp,
+                                su2double initial_value = 0);
```

Comment:
Just a note, default values don't really work with virtual functions (the one that matters is the base, usually)